RFC-007-SWAP-Ethereum-Basic.md

```diff
@@ -0,0 +1,291 @@
+```
+
+To compile the contract code replace the placeholder values with the HTLC paramters at the following offsets:
+
+
+| Data              | Position of first byte | Position of last byte | Length |
+|:------------------|:-----------------------|:----------------------|:-------|
+| `secret_hash`     | 51                     | 82                    | 32     |
+| `expiry`          | 99                     | 102                   | 4      |
+| `redeem_identity` | 153                    | 172                   | 20     |
+| `refund_identity` | 219                    | 248                   | 20     |
+
+The contract emits two logs:
+
+| Topic        | keccack256(Topic)                                                  | Data     | Emitted When             |
+|:-------------|:-------------------------------------------------------------------|----------|:-------------------------|
+| `Redeemed()` | 0xb8cac300e37f03ad332e581dea21b2f0b84eaaadc184a295fef71e81f44a7413 | `secret` | The contract is redeemed |
+| `Refunded()` | 0x5d26862916391bf49478b2f5103b0720a842b45ef145a268f2cd1fb2aed55178 | -        | The contract is refunded |
+
+
+Implementations SHOULD use the following gas limits on the transactions related to the contract:
+
+| Transaction | recommended gas limit |
+|:------------|-----------------------|
+| Deployment  | 121,000               |
+| Redeem      | 100,000               |
+| Refund      | 100,000               |
+
+## Execution Phase
+
+The following section describes how both parties should interact with the Ethereum blockchain during the [RFC003 execution phase](./RFC-003-SWAP-basic.md#execution-phase).
+
+### Deployment
+
+At the start of the deployment stage, both parties compile the contract code as described in the previous section.
+We will call this value `contract_code`.
+
+To deploy the Ether HTLC, the *funder* must deploy the `contract_code`.
+They SHOULD do this by sending a contract deployment transaction to the relevant Ethereum blockchain with:
+
+- `contract_code` as the `data` of the transaction
+- The quantity of wei specified in the Ether Asset header as the `value` of the transaction
+
+The funder SHOULD NOT do this executing the `CREATE` opcode from another contract.
+
+To be notified of the deployment event, both parties MAY watch the blockchain for a transaction with the `contract_code` as the data.
+Upon observing the deployment transaction, both parties SHOULD record the address the contract was deployed to (referred to as `contract_address` from now on).
+
+### Redeem
+
+Before redeeming, the redeemer SHOULD wait until the deployment transaction has enough confirmation such that they consider it permanent.
+
+To redeem the HTLC, the redeemer SHOULD send a transaction to the `contract_address` with the `data` of the transaction set to the `secret`.
```

Comment:
```suggestion
To redeem the HTLC, the redeemer sends a transaction to the `contract_address` with the `data` of the transaction set to the `secret`.
```

Reply:
Why don't you want the SHOULD here? We're telling them how they SHOULD redeem it. They may redeem it by calling the contract from another contract.